Dockerfile

```diff
@@ -32,32 +33,49 @@ RUN apt-get update && \
         ca-certificates \
         libreadline8 \
         python3 \
-        shared-mime-info && \
+        shared-mime-info  \
+    && \
+    rm -rf /var/lib/apt/lists/*
+
+COPY Gemfile* package.json yarn.lock /opt/mastodon/
```

Comment:
I probably would move the `WORKDIR` line down beside this `COPY Gemfile` line? The `RUN apt-get` doesn't depend on it.

Presumably the same with the `COPY --link --from=ruby` and ENV. Only `DEBIAN_FRONTEND` is relevant for the `RUN apt-get`? In that case you can leave that but change it to `ARG` if you don't need it to persist after building the stage (_more relevant to the `prod` stage as that builds the final image, but you've just inlined that into the `RUN` which is fine AFAIK_).

Reply:
I changed the order, thanks!